File: test_fb.go
package main
import (
	"bytes"
	"fmt"
	"io"
	"mime/multipart"
	"net/http"
	"regexp"
)
func main() {
	cookie := "datr=6M1JaPhd1g6T7V1vZuvpmKC0;xs=1%3A0FwF4gkw-dE3cw%3A2%3A1776045645%3A-1%3A-1;sb=Q07cafyMWDeguux1Y8IExiSZ;fr=1xDujY5gF4655mvfN.AWfs8C-gp2bpncuY2lxPJ3mDzYxyduoF6mR7BaUM96ko8B9pJKw.BobKSj..AAA.0.0.Bp3E5E.AWeSbM-PkJXpFO10lVymdjbJwdY;c_user=61571360758847;pas=61571360758847%3AhAIRYYu4Yh;locale=en_US"
	reqGet, _ := http.NewRequest("GET", "https://www.facebook.com/", nil)
	reqGet.Header.Set("Cookie", cookie)
	reqGet.Header.Set("User-Agent", "Mozilla/5.0")
	respGet, _ := http.DefaultClient.Do(reqGet)
	bodyBytes, _ := io.ReadAll(respGet.Body)
	respGet.Body.Close()
	htmlText := string(bodyBytes)
	fbDtsg := ""
	if m := regexp.MustCompile(`\["DTSGInitialData",\[\],{"token":"([^"]+)"}`).FindStringSubmatch(htmlText); len(m) > 1 {
		fbDtsg = m[1]
	}
	fmt.Println("DTSG: ", fbDtsg)
	
	uid := "61571360758847"

	// test composer photo/upload endpoint with video extension
	url := fmt.Sprintf("https://upload.facebook.com/ajax/react_composer/attachments/photo/upload?__a=1&__req=1&__user=%s", uid)
	body := &bytes.Buffer{}
	writer := multipart.NewWriter(body)
	writer.WriteField("fb_dtsg", fbDtsg)
	
	// mock file Upload
	part, _ := writer.CreateFormFile("farr", "test.mp4")
	part.Write([]byte("fake video data fake video data fake video data"))
	writer.Close()
	
	req, _ := http.NewRequest("POST", url, body)
	req.Header.Set("Cookie", cookie)
	req.Header.Set("Content-Type", writer.FormDataContentType())
	req.Header.Set("User-Agent", "Mozilla/5.0")
	resp, _ := http.DefaultClient.Do(req)
	r, _ := io.ReadAll(resp.Body)
	fmt.Println(resp.StatusCode)
	fmt.Println(string(r))
}

File: testfb.py
import requests
import json
import re

cookie = "datr=6M1JaPhd1g6T7V1vZuvpmKC0;xs=1%3A0FwF4gkw-dE3cw%3A2%3A1776045645%3A-1%3A-1;sb=Q07cafyMWDeguux1Y8IExiSZ;fr=1xDujY5gF4655mvfN.AWfs8C-gp2bpncuY2lxPJ3mDzYxyduoF6mR7BaUM96ko8B9pJKw.BobKSj..AAA.0.0.Bp3E5E.AWeSbM-PkJXpFO10lVymdjbJwdY;c_user=61571360758847;pas=61571360758847%3AhAIRYYu4Yh;locale=en_US"
uid = "61571360758847"
doc_id = "26339944415655330"

session = requests.Session()
session.headers.update({
    "cookie": cookie,
    "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64)"
})

print("Fetching profile to get fb_dtsg...")
resp = session.get(f"https://www.facebook.com/profile.php?id={uid}")
html = resp.text

# Extract fb_dtsg
match = re.search(r'"DTSGInitialData",\[\],\{"token":"([^"]+)"', html)
if not match:
    print("Could not find fb_dtsg in HTML.")
    exit(1)
fb_dtsg = match.group(1)
print("fb_dtsg:", fb_dtsg)

URL = "https://www.facebook.com/api/graphql/"
variables = {
    "count": 20,
    "id": uid,
    "scale": 1
}

payload = {
    "av": uid,
    "__user": uid,
    "__a": "1",
    "fb_dtsg": fb_dtsg,
    "variables": json.dumps(variables),
    "fb_api_req_friendly_name": "ProfileCometAppSectionFeedPaginationQuery"
}

print("Executing GraphQL request...")
post_resp = session.post(URL, data=payload)
try:
    data = post_resp.json()
    with open("python_debug_fb.json", "w", encoding="utf-8") as f:
        json.dump(data, f, indent=2, ensure_ascii=False)
    print("Saved response to python_debug_fb.json")
except Exception as e:
    print("Failed to parse JSON:", post_resp.text[:500])
